internal/command/commands.go

```diff
@@ -10,18 +10,17 @@ import (
 // Commands stores a slice of commands and provides some helper execution methods.
 type Commands []*Command
 
-// execute runs each command in the calling slice sequentially using the passed config and the outputs accumulated to that point.
-func (c Commands) execute(ctx context.Context, config *Config, args Args, previousOutputs Outputs, streams *genericclioptions.IOStreams) (outputs Outputs, err error) {
-	outputs = Outputs{}
-	for k, v := range previousOutputs {
-		outputs[k] = v
-	}
-
+// Execute runs each command in the calling slice sequentially using the passed config and the outputs accumulated to that point.
+func (c Commands) Execute(ctx context.Context, config *Config, args Args, outputs Outputs, streams *genericclioptions.IOStreams) (Outputs, error) {
```

Comment:
Is there any technical reason to make these public for testing vs testing them as private funcs? Not really talking about Execute specifically, it makes sense to have a public execute func on a public Command struct in its own right.

Reply:
Yes, the idea is that tests should cover the public API of a package and generally avoid testing un-exported functions. 

https://github.com/maratori/testpackage

Even if you don't use a separate test package it's still a good general rule to focus tests on the public API for a package and test unexported funcs only when necessary.

In this specific case I'm also anticipating ending up with `package commands` that handles command templating, output passing, and execution. `package execforward` will declare the hook structure and need to use both `commands.Commands` (for hooks) and `commands.Command` (for the main command).